### PR TITLE
docs: add ldap auth method

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
docs

### Description
This PR adds docs for the ldap auth method.

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/content/config/index.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ more secure authentication like 2FA or WebAuthn.
 | `HD_AUTH_LOCAL_ENABLE_REGISTER`           | `false` | `true`, `false`         | This makes it possible to register new local accounts in HedgeDoc.                    |
 | `HD_AUTH_LOCAL_MINIMAL_PASSWORD_STRENGTH` | `2`     | `0`, `1`, `2`, `3`, `4` | The minimum [zxcvbn-ts][zxcvbn-ts-score] password score, that passwords need to have. |
 
-**Password score ([zxcvbn-ts][zxcvbn-ts-score])**
+#### Password score ([zxcvbn-ts][zxcvbn-ts-score])
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:77]: The links shouldn't be present in the headline

File: docs/content/config/index.md [hunk 2]
```diff
@@ -84,6 +84,26 @@ more secure authentication like 2FA or WebAuthn.
...
 
+### LDAP
+
+HedgeDoc can use an LDAP server to authenticate users. As HedgeDoc supports multiple LDAP servers, you first need to tell HedgeDoc the servers you want to configure (`HD_AUTH_LDAPS`), and then you need to provide a few variables for those LDAP servers depending on how you want to use them.
+Each of those variables will contain the given name for this LDAP server. For example if you named your ldap server `MY_LDAP` all variables for this server will start with `HD_AUTH_LDAP_MY_LDAP`.
+
+| environment variable                       | default              | example                                            | description                                                                                             |
+|--------------------------------------------|----------------------|----------------------------------------------------|---------------------------------------------------------------------------------------------------------|
+| `HD_AUTH_LDAPS`                            | -                    | `MY_LDAP`                                          | A list of LDAP servers HedgeDoc should use, comma-seperated.                                            |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:94]: A comma-separated list of ....

File: docs/content/config/index.md [hunk 3]
```diff
@@ -84,6 +84,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_PROVIDER_NAME`         | `LDAP`               | `My LDAP`                                          | The name for the ldap server `$NAME`, that is shown in the UI of HegdeDoc.                              |
+| `HD_AUTH_LDAP_$NAME_URL`                   | -                    | `https://ldap.example.com`                         | The url with which the LDAP server `$NAME` can be accessed.                                             |
+| `HD_AUTH_LDAP_$NAME_SEARCH_BASE`           | -                    | `ou=users,dc=ldap,dc=example,dc=com`               | Where the user accounts are saved on the ldap server `$NAME`.                                           |
+| `HD_AUTH_LDAP_$NAME_SEARCH_FILTER`         | `(uid={{username}})` | `(&(uid={{username}})(objectClass=inetOrgPerson))` | Which user accounts should be accessible from the ldap server `$NAME`.                                  |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:98]: That's also no sentence. 

"A LDAP search filter that filters the users that should have access"

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]
Either you turn the explanation into proper sentences or you rephrase them shorter so they dont sound like subordinate clauses.

File: docs/content/config/index.md (hunk 1 above)
Comment (DerMolly) [reply docs/content/config/index.md:77]: What would you recommend instead? A short sentence beneath?
`The password score is calculated with [zxcvbn-ts][zxcvbn-ts-score].`

Comment (mrdrogdrog) [reply docs/content/config/index.md]: sounds good

File: docs/content/config/index.md [hunk 4]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
 |   3   | `safely unguessable` and `very unguessable` passwords are allowed | 100.000.000                                       |
 |   4   | Only `very unguessable` passwords are allowed                     | 10.000.000.000                                    |
 
+### LDAP
+
+HedgeDoc can use an LDAP server to authenticate users. As HedgeDoc supports multiple LDAP servers, you first need to tell HedgeDoc the servers you want to configure (`HD_AUTH_LDAPS`), and then you need to provide a few variables for those LDAP servers depending on how you want to use them.
+Each of those variables will contain the given name for this LDAP server. For example if you named your ldap server `MY_LDAP` all variables for this server will start with `HD_AUTH_LDAP_MY_LDAP`.
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:92]: You already explain here how the server naming works. imho you can remove the mentions of `the ldap server $NAME` in the descriptions below. That makes it a bit easier to read them.

File: docs/content/config/index.md [hunk 5]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
 |   3   | `safely unguessable` and `very unguessable` passwords are allowed | 100.000.000                                       |
 |   4   | Only `very unguessable` passwords are allowed                     | 10.000.000.000                                    |
 
+### LDAP
+
+HedgeDoc can use an LDAP server to authenticate users. As HedgeDoc supports multiple LDAP servers, you first need to tell HedgeDoc the servers you want to configure (`HD_AUTH_LDAPS`), and then you need to provide a few variables for those LDAP servers depending on how you want to use them.
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:91]:
```suggestion
HedgeDoc can use one or multiple LDAP servers to authenticate users. To do this, you first need to tell HedgeDoc the names of the servers you want to use (`HD_AUTH_LDAPS`) and then you need to provide the configuration variables for those LDAP servers depending on how you want to use them.
```

File: docs/content/config/index.md [hunk 6]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+Each of those variables will contain the given name for this LDAP server. For example if you named your ldap server `MY_LDAP` all variables for this server will start with `HD_AUTH_LDAP_MY_LDAP`.
+
+| environment variable                       | default              | example                                            | description                                                                                                     |
+|--------------------------------------------|----------------------|----------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|
+| `HD_AUTH_LDAPS`                            | -                    | `MY_LDAP`                                          | A comma-seperated list of LDAP servers HedgeDoc should use.                                                     |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:96]: You don't provide the LDAP servers here, you provide the NAMES of the servers.

File: docs/content/config/index.md [hunk 7]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_PROVIDER_NAME`         | `LDAP`               | `My LDAP`                                          | The name for the ldap server `$NAME`, that is shown in the UI of HegdeDoc.                                      |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:97]: I'll mark it only once: LDAP is an acronym and therefore should be written in uppercase.

Comment (mrdrogdrog) [reply docs/content/config/index.md:97]: To avoid confusion use the term "display name" in the description.

File: docs/content/config/index.md [hunk 8]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_URL`                   | -                    | `https://ldap.example.com`                         | The url with which the LDAP server `$NAME` can be accessed.                                                     |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:98]: LDAP urls don't use HTTP(s). They use `ldap://`, `ldaps://` or `ldapi://`

File: docs/content/config/index.md [hunk 9]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_SEARCH_BASE`           | -                    | `ou=users,dc=ldap,dc=example,dc=com`               | The ldap search base which contains the user accounts on the ldap server `$NAME`.                               |
+| `HD_AUTH_LDAP_$NAME_SEARCH_FILTER`         | `(uid={{username}})` | `(&(uid={{username}})(objectClass=inetOrgPerson))` | A ldap search filter that filters the users that should have access on the ldap server `$NAME`.                 |
+| `HD_AUTH_LDAP_$NAME_SEARCH_ATTRIBUTES`     | -                    | `username,cn`                                      | A comma-seperated list of attributes that the search filter from the ldap server `$NAME` should access.         |
+| `HD_AUTH_LDAP_$NAME_USERID_FIELD`          | `uid`                | `uid`, `uidNumber`, `sAMAccountName`               | Which field of the user account should be used as an id for the user on the ldap server `$NAME`.                |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:102]:
```suggestion
| `HD_AUTH_LDAP_$NAME_USERID_FIELD`          | `uid`                | `uid`, `uidNumber`, `sAMAccountName`               | The attribute which should be used as unique id for the user                |
```

File: docs/content/config/index.md [hunk 10]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_DISPLAY_NAME_FIELD`    | `displayName`        | `displayName`, `name`, `cn`                        | Which field of the user account should be used as the display name for the user on the the ldap server `$NAME`. |
+| `HD_AUTH_LDAP_$NAME_PROFILE_PICTURE_FIELD` | `jpegPhoto`          | `jpegPhoto`, `thumbnailPhoto`                      | Which field of the user account should be used as the user image for the user on the the ldap server `$NAME`.   |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:104]: see above

File: docs/content/config/index.md (hunk 9 above)
Comment (mrdrogdrog) [reply docs/content/config/index.md:102]: Don't start the descriptions with "which". That reads like a subordinate clause.

File: docs/content/config/index.md [hunk 11]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_BIND_DN`               | -                    | `cn=admin,dc=ldap,dc=example,dc=com`               | The dn to access the ldap server `$NAME`.                                                                       |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:105]:
```suggestion
| `HD_AUTH_LDAP_$NAME_BIND_DN`               | -                    | `cn=admin,dc=ldap,dc=example,dc=com`               | The dn which is used to perform the user search. If this is omitted then HedgeDoc will use an anonymous bind.                                                                       |
```

File: docs/content/config/index.md [hunk 12]
```diff
@@ -84,6 +86,26 @@ more secure authentication like 2FA or WebAuthn.
...
+| `HD_AUTH_LDAP_$NAME_BIND_CREDENTIALS`      | -                    | `MyLdapPassword`                                   | The credential to access the ldap server `$NAME`.                                                               |
+| `HD_AUTH_LDAP_$NAME_TLS_CERT_PATHS`        | -                    | `ldap-ca.pem`                                      | A comma-seperated list of TLS certificates for the ldap server `$NAME`.                                         |
```
Comment (mrdrogdrog) [inline docs/content/config/index.md:107]: You don't provide the TLS certificates. You provides paths to certificate files.

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: docs/content/config/index.md (hunk 8 above)
Comment (DerMolly) [reply docs/content/config/index.md:98]: Good point maybe we want to add this to the joi config for the LDAP url as well? 🤔

**Pull request merged**